CONTRIBUTING.md

```diff
@@ -495,6 +495,31 @@ In order to help the decision, CI automatically assigns [`rebuild` labels](https
 As a rule of thumb, if the number of rebuilds is **500 or more**, consider targeting the `staging` branch instead of `master`; if the number is **1000 or more**, the pull request causes a mass rebuild, and should target the `staging` branch.
 See [previously merged pull requests to the staging branches](https://github.com/NixOS/nixpkgs/issues?q=base%3Astaging+-base%3Astaging-next+is%3Amerged) to get a sense for what changes are considered mass rebuilds.
 
+Please note that changes to the Linux kernel are an exception to this rule. These PRs go to `batch-with-kernel-updates`,
+see [the next section for more context](#changes-rebuilding-all-tests).
+
+### Changes rebuilding all tests
+[test-driver-rebuild]: #changes-rebuilding-all-tests
+
+Changes causing a rebuild of all tests get a special [`10.rebuild-all-nixos-tests`](https://github.com/NixOS/nixpkgs/issues?q=state%3Aopen%20label%3A10.rebuild-nixos-tests) label.
+
+This means that all VM tests get rebuild which poses a significant impact on the build infrastructure.
+Hence, these PRs should either target a `staging`-branch or `batch-with-kernel-updates`, provided the following conditions apply:
+
+* The label `10.rebuild-all-nixos-tests` is set, or
+* The PR is a change affecting the Linux kernel.
+
+The branch gets merged weekly along with Linux kernel updates which directly go to `master` and also cause a rebuild of all tests.
+
+__Critical__ security updates that are qualified for `batch-with-kernel-updates` should be merged into this branch.
+After that, the entire branch should be merged into `master`.
+In exceptional cases where a lot of changes stacked up in `batch-with-kernel-updates` and a high-risk security release, the update can be directly shipped to `master`, short-circuiting the `batch-with-kernel-updates` branch.
```

Comment:
I don't really understand this part. Why would we ever short-circuit batch-with-kernel-updates?